packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2798,9 +2810,15 @@ class RenderFittedBox extends RenderProxyBox {
     );
   }
 
+  @override
+  bool paintsChild(RenderBox child) {
+    assert(child.parent == this);
+    return !size.isEmpty && !child.size.isEmpty;
+  }
+
   @override
   void applyPaintTransform(RenderBox child, Matrix4 transform) {
-    if (size.isEmpty || child.size.isEmpty) {
+    if (!paintsChild(child)) {
```

Comment:
Will this remove accessibility transform as well? I remember widget like Opacity can still include in semantics when opacity in zero. The semantics transform is calculate using this method

Reply:
This is existing behavior for this widget. Yes, this means the a11y transform will be zeroed out. If we want to change that I think it should be done in a separate PR.

Reply:
Oh nvm, this is RenderFittedBox, I thought it was RenderProxyBox